sensors/triggers/kafka/kafka.go

```diff
@@ -189,6 +193,19 @@ func (t *KafkaTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 		return nil, err
 	}
 
+	// Producer with avro schema
+	if trigger.SchemaRegistry != nil {
+		schema, err := getSchemaFromRegistry(trigger.SchemaRegistry)
```

Comment:
First: It is not frequently to change the scheme
Yes I agree. I think of 2 possible solutions:
- Cache (We need think about the key and ttl)
- Use schema raw (.avsc) file as file, this means we no longer need schema registry setting to get schema.

What do you think @whynowy

Reply:
cache is better,  if a schema changes it would generate new id anyway.  since schemaid is hardcoded in the config, I think can read the schema into a variable at start of the process

Reply:
Maybe we can put the `schema` into `KafkaTrigger` as a property, and init it in the New function. 

Today the trigger instance is newed per message, we can change it later to be cached.

Reply:
@whynowy @tooptoop4 , 

Using schema as a property maybe make sensor file much bigger,  usually the schema is bigger.  
Ex:  .avsc file has (64rows)
My sensor has (35rows)

I thinked to use a base64 of schema but I don't know if is a good ideia to reduce file size

My purpose is to use the raw file because it is more easy to control version and commits, and more easy to define and handling it.

Reply:
@whynowy would moving this code to eventsource mean it can cache response from schema registry?

as defining the avsc content means schema registry is not supported

Reply:
@Thomasr-02 - What I meant was not moving it to the sensor spec, but getting the schema with remote call in `NewKafkaTrigger()` function, and put the result as a private property in `KafkaTrigger` struct, in this way, for each instance of `KafkaTrigger`, there's only 1 time remote call to get the schema.

Today, every time when there's a need to trigger an action, we new a trigger instance (here, it's KafkaTrigger instance), see https://github.com/argoproj/argo-events/blob/master/sensors/trigger.go#L55, which means, with the approach above, we still do lots of remote call to get schemas, but this can be changed in a separate PR to cache the trigger instance.

@tooptoop4 - sensor and enventsource are totally decoupled, we should not move it to eventsource. Not quite clear about what you said about `as defining the avsc content means schema registry is not supported`.